src/coreir-c/coreir-c.cpp

```diff
@@ -108,6 +114,68 @@ extern "C" {
   }
 
 
+  bool CORECompileToVerilog(COREContext* ctx,
+                            COREModule* top,
+                            char* filename,
+                            int num_libs,
+                            char** libs,
+                            char* split,
+                            char* product,
+                            bool inline_,
+                            bool verilator_debug) {
+    auto context = reinterpret_cast<Context*>(ctx);
+    auto module = reinterpret_cast<Module*>(top);
+    context->setTop(module->getRefName());
+
+    // TODO(rsetaluri): This code duplicates much of the verilog compilation
+    // codepath in the coreir.cpp binary. We should extract this into a common
+    // place.
+    std::vector<std::string> libraries;
+    for (int i = 0; i < num_libs; i++) {
+      std::string lib(libs[i]);
+      context->getLibraryManager()->loadLib(lib);
+    }
+    // TODO(rsetaluri): Have option to output this or not.
+    CoreIRLoadVerilog_coreir(context);
+    CoreIRLoadVerilog_corebit(context);
+
+    std::string verilog_pass = "verilog";
+    if (inline_) verilog_pass += " -i";
+    if (verilator_debug) verilog_pass += " -y";
+    std::vector<std::string> namespaces {"global"};
+    std::vector<std::string> passes {
+      "rungenerators",
+      "removebulkconnections",
+      "flattentypes",
+      verilog_pass
+    };
+    context->runPasses(passes, namespaces);
+    auto pass = static_cast<Passes::Verilog*>(
+        context->getPassManager()->getAnalysisPass("verilog"));
+    defer(pass->clear());
```

Comment:
Why are you deferring this clear function?

Reply:
so that it always gets called. otherwise I would have to make sure to have it before every `exit()` point.

Reply:
e.g. L167, L174, L177